tests/unit/updateTestResults.unitTest.ts

```diff
@@ -0,0 +1,173 @@
+import {TestResultsService} from "../../src/services/TestResultsService";
+import {HTTPError} from "../../src/models/HTTPError";
+import fs from "fs";
+import path from "path";
+import {MESSAGES} from "../../src/assets/Enums";
+import {cloneDeep} from "lodash";
+import {ITestResult} from "../../src/models/ITestResult";
+
+describe("updateTestResults", () => {
+    let testResultsService: TestResultsService | any;
+    let MockTestResultsDAO: jest.Mock;
+    let testResultsMockDB: any;
+    const msUserDetails = {
+        msUser: "dorel",
+        msOid: "123456"
+    };
+    beforeEach(() => {
+        testResultsMockDB = JSON.parse(fs.readFileSync(path.resolve(__dirname, "../resources/test-results.json"), "utf8"));
+        MockTestResultsDAO = jest.fn().mockImplementation(() => {
+            return {};
+        });
+        testResultsService = new TestResultsService(new MockTestResultsDAO());
+    });
+
+    afterEach(() => {
+        testResultsMockDB = null;
+        testResultsService = null;
+        MockTestResultsDAO.mockReset();
+    });
+
+    context("when trying to update a test-result", () => {
+        context("and the payload is valid", () => {
+            context("and the test-result is found", () => {
+                it("should return the updated test-result", () => {
+                    MockTestResultsDAO = jest.fn().mockImplementation(() => {
+                        return {
+                            updateTestResult: () => {
+                                return Promise.resolve({});
+                            },
+                            getBySystemNumber: () => {
+                                return Promise.resolve({
+                                    Items: Array.of(testResultsMockDB[1]),
+                                    Count: 1
+                                });
+                            }
+                        };
+                    });
+
+                    testResultsService = new TestResultsService(new MockTestResultsDAO());
+                    expect.assertions(5);
+                    const testToUpdate = testResultsMockDB[1];
+                    testToUpdate.countryOfRegistration = "gb";
```

Comment:
It seems that we are repeating this pattern multiple times before the assertions, is it worth moving that into a beforeEach block so we only have to write it once? Thoughts?

Reply:
true, moved to beforeEach